go/enclave/events/subscription_manager.go

```diff
@@ -71,9 +71,23 @@ func (s *SubscriptionManager) RemoveSubscription(id uuid.UUID) {
 	delete(s.subscriptions, id)
 }
 
-// FilterRelevantLogs filters out logs that are not subscribed to, and organises the logs by their subscribing ID.
-// It uses a state DB created from the rollup with the given hash to identify lifecycle vs user topics.
-func (s *SubscriptionManager) FilterRelevantLogs(logs []*types.Log, rollupHash common.L2RootHash) map[uuid.UUID][]*types.Log {
+// FilteredLogs filters out irrelevant logs.
+func (s *SubscriptionManager) FilteredLogs(logs []*types.Log, rollupHash common.L2RootHash, account *gethcommon.Address) []*types.Log {
+	allLogs := []*types.Log{}
+	stateDB := s.storage.CreateStateDB(rollupHash)
+
+	for _, log := range logs {
+		userAddrs := getUserAddrs(log, stateDB)
```

Comment:
If this stuff becomes a hotspot I guess we might want to cache subscribedAddresses -> relevantLogs for recent blocks. But premature optimisation for now. Not sure how many transactions will be in a rollup and how many addresses in each but this bit could become non-negligible.

Reply:
Yeah. All stuff for the future IMO, at least once it's functionally ready.